lib/Document.php

```diff
@@ -221,14 +211,14 @@ public function createProperty($name, $value = null, array $parameters = null, $
         if (is_null($class)) {
             // If a VALUE parameter is supplied, we should use that.
             if (isset($parameters['VALUE'])) {
-                $class = $this->getClassNameForPropertyValue($parameters['VALUE']);
-                if (is_null($class)) {
-                    if (in_array(strtoupper($parameters['VALUE']), $this->allowedIllegalValues, true)) {
-                        unset($parameters['VALUE']);
-                        $class = $this->getClassNameForPropertyName($name);
-                    } else {
-                        throw new InvalidDataException('Unsupported VALUE parameter for '.$name.' property. You supplied "'.$parameters['VALUE'].'"');
-                    }
+                try {
+                    $class = $this->getClassNameForPropertyValue($parameters['VALUE']);
```

Comment:
Does it throw an exception or does it return null when it's invalid?

Reply:
the exception came from the invalid data are parsed as array, and when array is passed into the strtoupper function inside getClassNameForPropertyValue, we get an exception, so try-catch really is just to make sure we don't break under any random input, and for string type invalid data, getClassNameForPropertyValue would return null

Reply:
If not necessary for this ticket I would avoid catching the error and just handle the null case :sweat_smile:

Reply:
well to remove the try-catch I need to add a is_string check otherwise the test data we're using will break in strtoupper by throwing exception because it's actually array not string, which is due to the invalid data from the problematic ICS we received from user